pkg/config/transport_test.go

```diff
@@ -0,0 +1,203 @@
+		  "type": "object",
+		  "properties": {
+		  	"app": {
+		  		"type": "object",
+		  		"properties": {
+		  			"version": {
+		  				"description": "The version of the client application.",
+		  				"type": "string"
+		  			}
+		  		},
+		  		"required": ["version"]
+		  	}
+		  },
+		  "required": ["app"]
+	}`
+)
+
+var testInputs = []string{
+	`{}`,                                                                                                                                                                   // App missing
+	`{"app": {}}`,                                                                                                                                                          // Empty App object
+	`{"app": {"version": "6.4.1"}}`,                                                                                                                                        // Device missing
+	`{"app": {"version": "6.4.1"}, "device": {}, "os": {"platform": "WatchOS", "version": "9.4"}}`,                                                                         // Empty Device object
+	`{"app": {"version": "6.4.1"}, "device": {"location": {}, "os": {"platform": "WatchOS", "version": "9.4"}}}`,                                                           // Empty Location object
+	`{"app": {"version": "6.4.1"}, "device": {"location": {"locale": "en_GB"}, "os": {"platform": "WatchOS", "version": "9.4"}}}`,                                          // TimezoneOffset missing
+	`{"app": {"version": "6.4.1"}, "device": {"location": {"locale": "en_GB", "timezoneOffset": 7200}}}`,                                                                   // OS missing
+	`{"app": {"version": "6.4.1"}, "device": {"location": {"locale": "en_GB", "timezoneOffset": 7200}, "os": {}}}`,                                                         // Empty Os object
+	`{"app": {"version": "6.4.1"}, "device": {"location": {"locale": "en_GB", "timezoneOffset": 7200}, "os": {"platform": "WatchOS"}}}`,                                    // Version missing
+	`{"app": {"version": "6.4.1"}, "device": {"location": {"locale": "testLocale", "timezoneOffset": 7201}, "os": {"platform": "WatchOS", "version": "9.4"}}}`,             // Wrong locale format
+	`{"app": {"version": "6.4.1"}, "device": {"location": {"locale": "testLocale", "timezoneOffset": 7201}, "os": {"platform": "WatchOS", "version": "9.4"}}, "user": {}}`, // Empty User object
+}
+
+func TestPayloadValidation(t *testing.T) {
+
+	schema := gojsonschema.NewStringLoader(testSchema)
+
+	var (
+		want = "invalid JSON error"
+	)
+
+	for _, input := range testInputs {
+		json := gojsonschema.NewStringLoader(input)
+
+		result, err := gojsonschema.Validate(schema, json)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if result.Valid() {
+			t.Errorf("have %v, want %v", input, want)
+		}
+	}
+}
+
+func TestDecodeJSONSchemaEmptyBody(t *testing.T) {
```

Comment:
Great coverage on the possible failures. We should also have a test case that succeeds so we are confident the middleware works for the positive case.